abi/known/tonstakers.json

```diff
@@ -0,0 +1,1245 @@
+            "stack_type": "cell",
+            "format": "cell"
+          },
+          {
+            "name": "projected_total_balance",
+            "stack_type": "int",
+            "format": "int"
+          },
+          {
+            "name": "projected_supply",
+            "stack_type": "int",
+            "format": "int"
+          }
+        ]
+      },
+      {
+        "name": "get_controller_address",
+        "return_values": [
+          {
+            "name": "controller_address",
+            "stack_type": "slice",
+            "format": "addr"
+          }
+        ],
+        "parameters": [
+          {
+            "name": "controller_id",
+            "stack_type": "int",
+            "format": "int"
+          },
+          {
+            "name": "validator",
+            "stack_type": "slice",
+            "format": "addr"
+          }
+        ]
+      },
+      {
+        "name": "get_controller_address_legacy",
+        "return_values": [
+          {
+            "name": "workchain",
+            "stack_type": "int",
+            "format": "int"
+          },
+          {
+            "name": "addr_hash",
+            "stack_type": "int",
+            "format": "int"
+          }
+        ],
+        "parameters": [
+          {
+            "name": "controller_id",
+            "stack_type": "int",
+            "format": "int"
+          },
+          {
+            "name": "wc",
+            "stack_type": "int",
+            "format": "int"
+          },
+          {
+            "name": "addr_hash",
+            "stack_type": "int",
+            "format": "int"
+          }
+        ]
+      },
+      {
+        "name": "get_loan",
+        "return_values": [
+          {
+            "name": "borrowed_amount",
+            "stack_type": "int",
+            "format": "int"
+          },
+          {
+            "name": "accounted_interest",
+            "stack_type": "int",
+            "format": "int"
+          }
+        ],
+        "parameters": [
+          {
+            "name": "controller_id",
+            "stack_type": "int",
+            "format": "int"
+          },
+          {
+            "name": "validator_address",
+            "stack_type": "slice",
+            "format": "addr"
+          },
+          {
+            "name": "prev",
+            "stack_type": "int",
+            "format": "int",
+            "optional": true
+          },
+          {
+            "name": "update",
+            "stack_type": "int",
+            "format": "int",
+            "optional": true
+          }
+        ]
+      },
+      {
+        "name": "get_controller_loan_position",
+        "return_values": [
+          {
+            "name": "distance_to_median",
+            "stack_type": "int",
+            "format": "int"
+          },
+          {
+            "name": "expected",
+            "stack_type": "int",
+            "format": "int"
+          }
+        ],
+        "parameters": [
```

Comment:
There is no field `parameters`. You can use [json schema](https://github.com/tonindexer/anton/blob/main/abi/abi.schema.json).